File: src/gwt/src/org/rstudio/studio/client/workbench/views/chat/ChatConstants.java
/*
 * ChatConstants.java
 *
 * Copyright (C) 2025 by Posit Software, PBC
 *
 * Unless you have received this program directly from Posit Software pursuant
 * to the terms of a commercial license agreement with Posit Software, then
 * this program is licensed to you under the terms of version 3 of the
 * GNU Affero General Public License. This program is distributed WITHOUT
 * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
 * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
 * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
 *
 */
package org.rstudio.studio.client.workbench.views.chat;

public interface ChatConstants extends com.google.gwt.i18n.client.Messages {
    String chatTitle();
    String chatTabLabel();
    String checkingInstallationMessage();
    String chatNotInstalledMessage();
    String startingChatMessage();
    String restartingChatMessage();
    String chatUpdating();
    String chatUpdateComplete();
    String chatUpdateFailed(String error);
    String chatUpdateCheckFailed();
    String chatRetry();
    String chatDismiss();
    String chatIgnore();
    String chatNotInstalledTitle();
    String chatNotInstalledWithVersionMessage(String version);
    String chatNotInstalledDescription();
    String chatLearnMore();
    String chatInstallButton();
    String chatUpdateAvailableTitle();
    String chatUpdateAvailableWithVersionsMessage(String currentVersion, String newVersion);
    String chatUpdateButton();
    String chatIncompatibleVersion();
    String chatErrorPrefix(String error);
    String chatUpdateRequiredTitle();
    String chatRStudioTooOldMessage();
    String chatAssistantTooOldMessage();
    String chatProcessExitedTitle();
    String chatProcessExitedMessage();
    String chatRestartButton();
    String chatBackendStartFailed(String error);
    String chatBackendStartError(String error);
    String chatUpdateTimeout();
    String chatUpdateNotStarted();
    String chatUpdateStatusUnknown(String status);
    String chatUpdateStatusCheckFailed(String error);
    String chatBackendStartTimeout();
    String chatBackendStatusCheckFailed(String error);
    String chatRestartFailed(String error);
    String chatAssistantNotEnabledTitle();
    String chatAssistantNotEnabledMessage();
    String chatGlobalOptionsButton();
}

File: src/cpp/session/modules/chat/ChatConstants.cpp
/*
 * ChatConstants.cpp
 *
 * Copyright (C) 2025 by Posit Software, PBC
 *
 * Unless you have received this program directly from Posit Software pursuant
 * to the terms of a commercial license agreement with Posit Software, then
 * this program is licensed to you under the terms of version 3 of the
 * GNU Affero General Public License. This program is distributed WITHOUT
 * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
 * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
 * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
 *
 */

#include "ChatConstants.hpp"

namespace rstudio {
namespace session {
namespace modules {
namespace chat {
namespace constants {

// Installation paths
const char* const kPositAiDirName = "pai/bin";
const char* const kClientDirPath = "dist/client";
const char* const kServerScriptPath = "dist/server/main.js";
const char* const kIndexFileName = "index.html";
const char* const kCspConfigPath = "dist/csp.json";

// Protocol Version (SUPPORTED_PROTOCOL_VERSION)
const char* const kProtocolVersion = "8.0";

// Capabilities: JSON-RPC methods that RStudio handles
const std::vector<std::string>& rstudioCapabilities()
{
   static const std::vector<std::string> s_capabilities = {
      "runtime/getActiveSession",
      "runtime/getDetailedContext",
      "runtime/executeCode",
      "runtime/getConsoleContent",
      "workspace/readFileContent",
      "workspace/writeFileContent",
      "workspace/editFileContent",
      "workspace/insertIntoNewFile",
      "workspace/insertAtCursor",
      "ui/openDocument",
   };
   return s_capabilities;
}

} // namespace constants
} // namespace chat
} // namespace modules
} // namespace session
} // namespace rstudio
